source/shared/cpp/ObjectModel/MarkDownBlockParser.cpp

```diff
@@ -451,45 +473,133 @@ void LinkParser::CaptureLinkToken()
 }
 
-// list has form of -\s+ markdown block and a line break
-// this method checks such syntax
+// list marker have form of ^-\s+ or \r-\s+
+// this method matches -\s
+bool ListParser::MatchNewListItem(std::stringstream &stream)
+{
+    if (IsHyphen(stream.peek()))
+    {
+        stream.get();
+        if (stream.peek() == ' ')
+        {
+            stream.unget();
+            return true;
+        }
+        stream.unget();
+    }
+    return false;
+}
+
+// if lines are seperated by more than two new lines,
+// they are new block items
+// caller of this method is expected to have matched new line char 
+// before calling this method
+// this method will return true, after it mataches new line char
+// at least once.
+bool ListParser::MatchNewBlock(std::stringstream &stream)
+{
+    if (IsNewLine(stream.peek()))
+    {
+        do
+        {
+            stream.get();
+        } while (IsNewLine(stream.peek()));
+
+        return true;
+    }
+
+    return false;
+}
+
+// ordered list marker has form of ^\d+\.\s* or [\r,\n]\d+\.\s*, and this method checks the syntax
+// this method matches \d+\.
+bool ListParser::MatchNewOrderedListItem(std::stringstream &stream, std::string &number_string)
+{
+    do
+    {
+        number_string += stream.get();
+    } while (isdigit(stream.peek()));
+
+    if (stream.peek() == '.')
+    {
+        // ordered list syntax check complete
+        stream.unget();
+        return true;
+    }
+
+    return false;
+}
+
+// parse blocks that wasn't captured
+// if what we encounter is one of following items, start of new list, list item, or new block element,
+// we do not include in the current block, we return, and have it handled by the caller
+void ListParser::ParseSubBlocks(std::stringstream &stream)
+{
+    while (!stream.eof()) 
+    { 
+        if (IsNewLine(stream.peek()))
+        {
+            int newLineChar = stream.get();
+            // check if it is the start of new block items
+            if (isdigit(stream.peek()))
+            {
+                std::string number_string = "";
+                if (MatchNewOrderedListItem(stream, number_string))
+                {
+                    break;
+                }
+                else
+                { 
+                    m_parsedResult.AddNewTokenToParsedResult(number_string);
+                }
+            }
+            else if (MatchNewListItem(stream) || MatchNewBlock(stream))
+            {
+                break;
+            }
+
+            m_parsedResult.AddNewTokenToParsedResult(newLineChar);
+        }
+        ParseBlock(stream);
+    }
+}
+// list marker has a form of ^-\s+ or [\r, \n]-\s+, and this method checks the syntax
 void ListParser::Match(std::stringstream &stream) 
 { 
-    if (stream.peek() == '-')
+    // check for the mandatory space
+    if (IsHyphen(stream.peek()))
     {
         stream.get();
-        // check for the mendatory space
         if (stream.peek() == ' ')
```

Comment:
>stream.peek() == ' ' [](start = 12, length = 20)

IsHyphen?

Reply:
you mean comment is not matching?
it is matching space after hyphen.

---
In reply to: [158163314](https://github.com/Microsoft/AdaptiveCards/pull/1053#discussion_r158163314) [](ancestors = 158163314)

Reply:
I meant, could you use the IsHyphen() function here?

---
In reply to: [158163641](https://github.com/Microsoft/AdaptiveCards/pull/1053#discussion_r158163641) [](ancestors = 158163641,158163314)

Reply:
i'm not sure... it's a space.

---
In reply to: [158164771](https://github.com/Microsoft/AdaptiveCards/pull/1053#discussion_r158164771) [](ancestors = 158164771,158163641,158163314)

Reply:
Ah, my bad, my codeflow is showing visible spaces so it looked like a hyphen to me, nevermind :)

---
In reply to: [158165040](https://github.com/Microsoft/AdaptiveCards/pull/1053#discussion_r158165040) [](ancestors = 158165040,158164771,158163641,158163314)